coffee-chats/src/main/java/com/google/step/coffee/servlets/AuthServlet.java

```diff
@@ -0,0 +1,38 @@
+package com.google.step.coffee.servlets;
+
+import com.google.step.coffee.HttpError;
+import com.google.step.coffee.JsonServlet;
+import com.google.step.coffee.PermissionChecker;
+import com.google.step.coffee.UserManager;
+import com.google.step.coffee.entity.User;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@WebServlet("/api/auth")
+public class AuthServlet extends JsonServlet {
+  private static class Response {
+    private String logoutUrl;
+    private User user;
+
+    public Response() {
+      this.logoutUrl = UserManager.getLogoutUrl();
+      this.user = UserManager.getCurrentUser();
+    }
+
+    public String getLogoutUrl() {
+      return logoutUrl;
+    }
+
+    public User getUser() {
+      return user;
+    }
+  }
+
+  @Override
+  public Object get(HttpServletRequest request) throws IOException, HttpError {
+    PermissionChecker.ensureLoggedIn();
```

Comment:
If most of servlets are expected to call this, I'd probably put it to the base class, or e.g. separate class in the hierarchy which does it. With some way to disable it perhaps. e.g. via annotation or a virtual call.

Reply:
I mean, just to make it harder to forget to call it